diff --git a/node_modules/react-native-vector-icons/.DS_Store b/node_modules/react-native-vector-icons/.DS_Store
new file mode 100644
index 0000000..feb2dfe
Binary files /dev/null and b/node_modules/react-native-vector-icons/.DS_Store differ
diff --git a/node_modules/react-native-vector-icons/RNVectorIcons.podspec b/node_modules/react-native-vector-icons/RNVectorIcons.podspec
index 99d5b06..bddb179 100644
--- a/node_modules/react-native-vector-icons/RNVectorIcons.podspec
+++ b/node_modules/react-native-vector-icons/RNVectorIcons.podspec
@@ -13,7 +13,7 @@ Pod::Spec.new do |s|
   s.source         = { :git => package["repository"]["url"], :tag => "v#{s.version}" }
 
   s.source_files   = 'RNVectorIconsManager/**/*.{h,m,mm,swift}'
-  s.resources      = "Fonts/*.ttf"
+#   s.resources      = "Fonts/*.ttf"
   s.preserve_paths = "**/*.js"
   # React Native Core dependency
   if defined? install_modules_dependencies
